<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Form</class>
 <widget class="QWidget" name="Form">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>690</width>
    <height>809</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_23">
   <item>
    <widget class="QTabWidget" name="fuzzy_modeling_tabs">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="fuzzy_membership_tab">
      <attribute name="title">
       <string>Fuzzy memberships</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_7">
       <item>
        <widget class="QScrollArea" name="scrollArea_3">
         <property name="widgetResizable">
          <bool>true</bool>
         </property>
         <widget class="QWidget" name="scrollAreaWidgetContents_3">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>0</y>
            <width>648</width>
            <height>740</height>
           </rect>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_8">
           <item>
            <layout class="QFormLayout" name="formLayout_7">
             <item row="0" column="0">
              <widget class="QLabel" name="label_3">
               <property name="text">
                <string>Input raster</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QgsMapLayerComboBox" name="input_raster_membership"/>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="label_14">
               <property name="text">
                <string>Output raster</string>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QLabel" name="membership_type_label">
               <property name="text">
                <string>Membership type</string>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QComboBox" name="membership_type">
               <item>
                <property name="text">
                 <string>Gaussian</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Large</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Linear</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Near</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Power</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Small</string>
                </property>
               </item>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QgsFileWidget" name="output_raster_membership">
               <property name="fileWidgetButtonVisible">
                <bool>true</bool>
               </property>
               <property name="storageMode">
                <enum>QgsFileWidget::SaveFile</enum>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QGroupBox" name="membership_parameters_groupbox">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>152</height>
              </size>
             </property>
             <property name="title">
              <string>Parameters</string>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_2">
              <item>
               <widget class="QStackedWidget" name="membership_parameters_pages">
                <property name="currentIndex">
                 <number>0</number>
                </property>
                <widget class="QWidget" name="gaussian_page">
                 <layout class="QFormLayout" name="formLayout">
                  <item row="0" column="0">
                   <widget class="QLabel" name="label">
                    <property name="text">
                     <string>Midpoint</string>
                    </property>
                   </widget>
                  </item>
                  <item row="0" column="1">
                   <widget class="QgsDoubleSpinBox" name="gaussian_function_midpoint">
                    <property name="decimals">
                     <number>3</number>
                    </property>
                    <property name="maximum">
                     <double>9999.000000000000000</double>
                    </property>
                    <property name="value">
                     <double>10.000000000000000</double>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="0">
                   <widget class="QLabel" name="label_2">
                    <property name="text">
                     <string>Spread</string>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="1">
                   <widget class="QgsDoubleSpinBox" name="gaussian_function_spread">
                    <property name="decimals">
                     <number>3</number>
                    </property>
                    <property name="maximum">
                     <double>9999.000000000000000</double>
                    </property>
                    <property name="value">
                     <double>0.010000000000000</double>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </widget>
                <widget class="QWidget" name="large_page">
                 <layout class="QFormLayout" name="formLayout_2">
                  <item row="0" column="0">
                   <widget class="QLabel" name="label999">
                    <property name="text">
                     <string>Midpoint</string>
                    </property>
                   </widget>
                  </item>
                  <item row="0" column="1">
                   <widget class="QgsDoubleSpinBox" name="large_function_midpoint">
                    <property name="decimals">
                     <number>3</number>
                    </property>
                    <property name="minimum">
                     <double>-9999.000000000000000</double>
                    </property>
                    <property name="maximum">
                     <double>9999.000000000000000</double>
                    </property>
                    <property name="value">
                     <double>50.000000000000000</double>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="0">
                   <widget class="QLabel" name="label_4">
                    <property name="text">
                     <string>Spread</string>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="1">
                   <widget class="QgsDoubleSpinBox" name="large_function_spread">
                    <property name="decimals">
                     <number>3</number>
                    </property>
                    <property name="maximum">
                     <double>9999.000000000000000</double>
                    </property>
                    <property name="value">
                     <double>5.000000000000000</double>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </widget>
                <widget class="QWidget" name="linear_page">
                 <layout class="QFormLayout" name="formLayout_3">
                  <item row="0" column="0">
                   <widget class="QLabel" name="label_5">
                    <property name="text">
                     <string>Lower bound</string>
                    </property>
                   </widget>
                  </item>
                  <item row="0" column="1">
                   <widget class="QgsDoubleSpinBox" name="linear_low_bound">
                    <property name="decimals">
                     <number>3</number>
                    </property>
                    <property name="minimum">
                     <double>-9999.000000000000000</double>
                    </property>
                    <property name="maximum">
                     <double>9999.000000000000000</double>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="0">
                   <widget class="QLabel" name="label_6">
                    <property name="text">
                     <string>Upper bound</string>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="1">
                   <widget class="QgsDoubleSpinBox" name="linear_high_bound">
                    <property name="decimals">
                     <number>3</number>
                    </property>
                    <property name="minimum">
                     <double>-9999.000000000000000</double>
                    </property>
                    <property name="maximum">
                     <double>9999.000000000000000</double>
                    </property>
                    <property name="value">
                     <double>1.000000000000000</double>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </widget>
                <widget class="QWidget" name="near_page">
                 <layout class="QFormLayout" name="formLayout_4">
                  <item row="0" column="0">
                   <widget class="QLabel" name="label_7">
                    <property name="text">
                     <string>Midpoint</string>
                    </property>
                   </widget>
                  </item>
                  <item row="0" column="1">
                   <widget class="QgsDoubleSpinBox" name="near_function_midpoint">
                    <property name="decimals">
                     <number>3</number>
                    </property>
                    <property name="minimum">
                     <double>-9999.000000000000000</double>
                    </property>
                    <property name="maximum">
                     <double>9999.000000000000000</double>
                    </property>
                    <property name="value">
                     <double>50.000000000000000</double>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="0">
                   <widget class="QLabel" name="label_8">
                    <property name="text">
                     <string>Spread</string>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="1">
                   <widget class="QgsDoubleSpinBox" name="near_function_spread">
                    <property name="decimals">
                     <number>3</number>
                    </property>
                    <property name="maximum">
                     <double>9999.000000000000000</double>
                    </property>
                    <property name="value">
                     <double>0.010000000000000</double>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </widget>
                <widget class="QWidget" name="power_page">
                 <layout class="QFormLayout" name="formLayout_5">
                  <item row="0" column="0">
                   <widget class="QLabel" name="label_9">
                    <property name="text">
                     <string>Lower bound</string>
                    </property>
                   </widget>
                  </item>
                  <item row="0" column="1">
                   <widget class="QgsDoubleSpinBox" name="power_low_bound">
                    <property name="decimals">
                     <number>3</number>
                    </property>
                    <property name="minimum">
                     <double>-9999.000000000000000</double>
                    </property>
                    <property name="maximum">
                     <double>9999.000000000000000</double>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="0">
                   <widget class="QLabel" name="label_10">
                    <property name="text">
                     <string>Upper bound</string>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="1">
                   <widget class="QgsDoubleSpinBox" name="power_high_bound">
                    <property name="decimals">
                     <number>3</number>
                    </property>
                    <property name="minimum">
                     <double>-9999.000000000000000</double>
                    </property>
                    <property name="maximum">
                     <double>9999.000000000000000</double>
                    </property>
                    <property name="value">
                     <double>1.000000000000000</double>
                    </property>
                   </widget>
                  </item>
                  <item row="2" column="0">
                   <widget class="QLabel" name="label_11">
                    <property name="text">
                     <string>Exponent</string>
                    </property>
                   </widget>
                  </item>
                  <item row="2" column="1">
                   <widget class="QgsDoubleSpinBox" name="power_function_exponent">
                    <property name="decimals">
                     <number>3</number>
                    </property>
                    <property name="maximum">
                     <double>9999.000000000000000</double>
                    </property>
                    <property name="value">
                     <double>2.000000000000000</double>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </widget>
                <widget class="QWidget" name="small_page">
                 <layout class="QFormLayout" name="formLayout_6">
                  <item row="0" column="0">
                   <widget class="QLabel" name="label_12">
                    <property name="text">
                     <string>Function midpoint</string>
                    </property>
                   </widget>
                  </item>
                  <item row="0" column="1">
                   <widget class="QgsDoubleSpinBox" name="small_function_midpoint">
                    <property name="decimals">
                     <number>3</number>
                    </property>
                    <property name="minimum">
                     <double>-9999.000000000000000</double>
                    </property>
                    <property name="maximum">
                     <double>9999.000000000000000</double>
                    </property>
                    <property name="value">
                     <double>50.000000000000000</double>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="0">
                   <widget class="QLabel" name="label_13">
                    <property name="text">
                     <string>Function spread</string>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="1">
                   <widget class="QgsDoubleSpinBox" name="small_function_spread">
                    <property name="decimals">
                     <number>3</number>
                    </property>
                    <property name="maximum">
                     <double>9999.000000000000000</double>
                    </property>
                    <property name="value">
                     <double>5.000000000000000</double>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </widget>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="btns_layout">
             <item>
              <widget class="QPushButton" name="reset_btn">
               <property name="text">
                <string>Reset</string>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <widget class="QPushButton" name="preview_btn">
               <property name="text">
                <string>Preview</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="run_membership_btn">
               <property name="text">
                <string>Run</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <spacer name="verticalSpacer_2">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QFrame" name="membership_plot_container">
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>350</height>
              </size>
             </property>
             <property name="frameShape">
              <enum>QFrame::StyledPanel</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Raised</enum>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="fuzzy_overlay_tab">
      <attribute name="title">
       <string>Fuzzy overlay</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QScrollArea" name="scrollArea">
         <property name="widgetResizable">
          <bool>true</bool>
         </property>
         <widget class="QWidget" name="scrollAreaWidgetContents">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>0</y>
            <width>648</width>
            <height>740</height>
           </rect>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_6">
           <item>
            <widget class="QGroupBox" name="fuzzy_rasters_box">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="title">
              <string>Rasters to overlay</string>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_3">
              <item>
               <layout class="QVBoxLayout" name="fuzzy_rasters_layout"/>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QGroupBox" name="fuzzy_method_box">
             <property name="title">
              <string>Fuzzy method</string>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_4">
              <item>
               <layout class="QVBoxLayout" name="fuzzy_method_layout">
                <item>
                 <widget class="QRadioButton" name="and_method">
                  <property name="text">
                   <string>And</string>
                  </property>
                  <property name="checked">
                   <bool>true</bool>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QRadioButton" name="or_method">
                  <property name="text">
                   <string>Or</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QRadioButton" name="sum_method">
                  <property name="text">
                   <string>Sum</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QRadioButton" name="product_method">
                  <property name="text">
                   <string>Product</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <layout class="QFormLayout" name="gamma_method_layout">
                  <item row="0" column="0">
                   <widget class="QRadioButton" name="gamma_method">
                    <property name="text">
                     <string>Gamma</string>
                    </property>
                   </widget>
                  </item>
                  <item row="0" column="1">
                   <widget class="QgsDoubleSpinBox" name="gamma_value">
                    <property name="maximumSize">
                     <size>
                      <width>150</width>
                      <height>16777215</height>
                     </size>
                    </property>
                    <property name="decimals">
                     <number>3</number>
                    </property>
                    <property name="maximum">
                     <double>1.000000000000000</double>
                    </property>
                    <property name="value">
                     <double>0.500000000000000</double>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QGroupBox" name="output_raster_box">
             <property name="title">
              <string>Output raster</string>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_5">
              <item>
               <widget class="QgsFileWidget" name="output_raster"/>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="run_overlay_btn">
             <property name="text">
              <string>Run</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="verticalSpacer">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>365</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QgsDoubleSpinBox</class>
   <extends>QDoubleSpinBox</extends>
   <header>qgsdoublespinbox.h</header>
  </customwidget>
  <customwidget>
   <class>QgsFileWidget</class>
   <extends>QWidget</extends>
   <header>qgsfilewidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QgsMapLayerComboBox</class>
   <extends>QComboBox</extends>
   <header>qgsmaplayercombobox.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
